Spigot-API-Patches/0300-Add-basic-ResourcePack-DataPack-API.patch

```diff
@@ -0,0 +1,131 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Connor Linfoot <connorlinfoot@me.com>
+Date: Sun, 16 May 2021 15:07:34 +0100
+Subject: [PATCH] Add basic ResourcePack/DataPack API
+
+
+diff --git a/src/main/java/io/papermc/paper/resource/ResourcePack.java b/src/main/java/io/papermc/paper/resource/ResourcePack.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..0583535bebb64a6a7aa46b15a97cf97be4496f89
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/resource/ResourcePack.java
+@@ -0,0 +1,26 @@
++package io.papermc.paper.resource;
++
++import org.checkerframework.checker.nullness.qual.NonNull;
++
++public interface ResourcePack {
++
++    /**
++     * @return the name of the pack
++     */
++    @NonNull
++    String getName();
++
++    /**
++     * @return the compatibility of the pack
++     */
++    @NonNull
++    ResourcePackCompatibility getCompatibility();
++
++    /**
++     * @return whether or not the pack is currently enabled
++     */
++    boolean isEnabled();
++
++    void setEnabled(boolean enabled);
++
++}
+diff --git a/src/main/java/io/papermc/paper/resource/ResourcePackCompatibility.java b/src/main/java/io/papermc/paper/resource/ResourcePackCompatibility.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..f0f6e28754df2c552c0f139ea9d7f50e65ad6a6a
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/resource/ResourcePackCompatibility.java
+@@ -0,0 +1,7 @@
++package io.papermc.paper.resource;
++
++public enum ResourcePackCompatibility {
```

Comment:
I'd probably throw this enum inside the ResourcePack interface, kinda small and specific for its own file.